src/main/java/loanbook/ui/LoanCard.java

```diff
@@ -56,14 +60,26 @@ public LoanCard(Loan loan, int displayedIndex) {
     private String getTagColorStyleFor(String tagName) {
         // we use the hash code of the tag name to generate a random color, so that the color remain consistent
         // between different runs of the program while still making it random enough between tags.
-        return TAG_COLOR_STYLES[Math.abs(tagName.hashCode()) % TAG_COLOR_STYLES.length];
+        switch (tagName) {
+        case "Ongoing":
+            return LOANSTATUS_TAG_COLOR_STYLES[2];
+        case "Returned":
+            return LOANSTATUS_TAG_COLOR_STYLES[1];
+        case "Deleted":
+            return LOANSTATUS_TAG_COLOR_STYLES[0];
+        default:
+            // All user defined tags are set to brown colour.
+            return TAG_COLOR_STYLES[3];
+        }
     }
 
     /**
      * Creates the tag labels for {@code loan}.
      */
     private void initTags(Loan loan) {
-        loan.getTags().forEach(tag -> {
+        HashSet<Tag> currentTags = new HashSet<>(loan.getTags());
+        currentTags.add(new Tag(loan.getLoanStatus().toString()));
```

Comment:
This is nice, but when there are other tags, the "Returned" or "Ongoing" tag may not be the first tag that appears. Not exactly a problem per se; but it might be nice to have it at the start.

![image](https://user-images.githubusercontent.com/35865440/48182162-31b5cc00-e365-11e8-9bd3-cd0fe8ec026d.png)

One way to do this is to change the code to the following:

```java
ArrayDeque<Tag> currentTags = new ArrayDeque<>(loan.getTags());
currentTags.addFirst(new Tag(loan.getLoanStatus().toString()));
```

![image](https://user-images.githubusercontent.com/35865440/48182228-688be200-e365-11e8-8c86-6e4bca7d6920.png)

Reply:
That's a good idea. However, changing this would cause the test cases to fail the GUI checks. In this case, I would suggest to push this minor upgrade to v2.0 instead.